### PR TITLE
feat: dev-schematics - introduce config merge mechanism

### PR DESCRIPTION
**Pull request by znikola**
This PR adds a mechanism that can merge the Spartacus configuration objects.
At the same time, it fixes a bug in dev-schematics where configuration is duplicated if it already exists in the app.module.ts

### CODE REVIEW

File: projects/dev-schematics/src/add-baseSites/index.ts [hunk 1]
```diff
@@ -1,24 +1,57 @@
...
-          'apparel-uk-spa',
-        ],
-      }`;
-    insertPropertyInStorefrontModuleCallExpression(
-      tree,
-      'src/app/app.module.ts',
-      insertion
+  return (tree: Tree, context: SchematicContext) => {
+    // TODO: is there a mechanism (an exposed method maybe) to find the app.module.ts without using the hard-coded path?
+    const appModulePath = 'src/app/app.module.ts';
```
Comment (psmul) [inline projects/dev-schematics/src/add-baseSites/index.ts]: Not sure about hardcoded path. I believe we had method in Schematics allowing us to find path for specified file.

Comment (psmul) [reply projects/dev-schematics/src/add-baseSites/index.ts]: add-spartacus/index.ts:200 we have method used: `getAppModulePath`

File: projects/dev-schematics/package.json [hunk 2]
```diff
@@ -15,13 +15,18 @@
   "author": "",
   "license": "MIT",
   "schematics": "./src/collection.json",
+  "peerDependencies": {
+    "@spartacus/storefront": "next",
```
Comment (psmul) [inline projects/dev-schematics/package.json]: Sorry, maybe I don't see something here, but why do we need this dep?

**Review by psmul — CHANGES_REQUESTED** [review]

File: projects/dev-schematics/src/add-baseSites/index.ts (hunk 1 above)
Comment (znikola) [reply projects/dev-schematics/src/add-baseSites/index.ts]: I found a way, there's `import { getAppModulePath } from '@schematics/angular/utility/ng-ast-utils';`

File: projects/dev-schematics/package.json (hunk 2 above)
Comment (znikola) [reply projects/dev-schematics/package.json]: It's not needed, I removed it.

File: projects/dev-schematics/package.json [hunk 3]
```diff
@@ -15,21 +15,22 @@
   "author": "",
   "license": "MIT",
   "schematics": "./src/collection.json",
+  "peerDependencies": {
+    "@spartacus/schematics": "next"
```
Comment (psmul) [inline projects/dev-schematics/package.json:19]: Should we define next here? What about final releases?

File: projects/dev-schematics/src/add-routing/index.ts [hunk 4]
```diff
@@ -1,8 +1,20 @@
-import { chain, Rule, Tree } from '@angular-devkit/schematics';
+import { chain, Rule, SchematicsException, Tree } from '@angular-devkit/schematics';
+import { getAppModulePath } from '@schematics/angular/utility/ng-ast-utils';
+import { getProjectTargets } from '@spartacus/schematics';
+import { Schema as SpartacusDevSchematicsOptions } from '../ng-add/schema';
 import { insertPropertyInStorefrontModuleCallExpression } from '../shared/utils/module-file-utils';
 
-function provideTestRouting(): Rule {
+function provideTestRouting(options: SpartacusDevSchematicsOptions): Rule {
```
Comment (psmul) [inline projects/dev-schematics/src/add-routing/index.ts:12]: `provideTestRouting` seems misleading, but can't remind myself reason behind this name. Maybe let's change it to more appropriate

Comment (znikola) [reply projects/dev-schematics/src/add-routing/index.ts:12]: Sure. Do you have any suggestion?

File: projects/dev-schematics/package.json (hunk 3 above)
Comment (znikola) [reply projects/dev-schematics/package.json:19]: I think the idea here is to always use the latest schematics - whether from npm, or from verdaccio (when testing locally).

File: projects/dev-schematics/src/add-routing/index.ts (hunk 4 above)
Comment (znikola) [reply projects/dev-schematics/src/add-routing/index.ts:12]: After an offline discussion, we decided to leave it as is.

**Review by psmul — APPROVED** [review]

**Pull request merged**